### DISCUSSION TITLE
I've no clue on how to manage CORS when I try to use the full stack hosted in a remote VM (i.e AWS)

### DISCUSSION THREAD

**Discussion by saphoooo** [realworld-apps/realworld#1147] [Q&A]
I'm trying to run the application (frontend and backend separately), but I'm struggle with CORS. And as I'm doing this to learn how to instrument each service for observability, and that I'm not a developper, I really don't know how to deal with that... Any help is more than welcome!

**Comment by geromegrignon** [comment]
Hi @saphoooo, which projects (front/back) are you using?

**Comment by saphoooo** [comment]
Doesn't really matter, I would like to be able to use as many as possible.
I think I tried an Angular frontend with a Java backend and then a Ruby
backend, with exactly the same result.

Le mer. 15 févr. 2023 à 09:21, Gerome Grignon ***@***.***> a
écrit :

**Reply by geromegrignon** [reply]
It matters as each library/framework has its own way to deal with cors. There is no out-of-the-box default configuration.

**Comment by saphoooo** [comment]
ok, fair enough. then, I'm currently working with this one:
https://github.com/alexeymezenin/ruby-on-rails-realworld-example-app

**Reply by geromegrignon** [reply]
I'm unfamiliar with Ruby, but this guide should provide you with related snippets to update the project: https://www.stackhawk.com/blog/rails-cors-guide/

It's related to this commented file: https://github.com/alexeymezenin/ruby-on-rails-realworld-example-app/blob/main/config/initializers/cors.rb

It'll require updating the `origins "example.com"` part to match the origin of your project on AWS side.
Don't hesitate to interact with the repository's author for further information.
